buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -686,8 +686,8 @@ class BuildPlugin implements Plugin<Project> {
                     test.jvmArgs '-ea', '-esa'
                 }
 
-                // we use './temp' since this is per JVM and tests are forbidden from writing to CWD
-                test.systemProperties 'java.io.tmpdir': './temp',
+                // we use 'temp' relative to CWD since this is per JVM and tests are forbidden from writing to CWD
+                test.systemProperties 'java.io.tmpdir': test.workingDir.toPath().resolve('temp'),
```

Comment:
Since we are changing this to an absolute path we need to ignore this system property for the purposes of task input snapshotting or it'll break cache reuse. We'll need to move this down below where we set other properties with absolute paths via `nonInputProperties`.

Reply:
I pushed 0dba342999386573fe98decb39e224e824f8ac1e.